bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/handler/VehicleHandler.java

```diff
@@ -153,21 +158,26 @@ private void queryApiAndUpdateChannels() {
         VolvoOnCallBridgeHandler bridgeHandler = getBridgeHandler();
         if (bridgeHandler != null) {
             try {
-                vehicleStatus = bridgeHandler.getURL(Status.class, configuration.vin);
+                vehicleStatus = bridgeHandler.getURL(vehicle.statusURL, Status.class);
                 vehiclePosition = new VehiclePositionWrapper(bridgeHandler.getURL(Position.class, configuration.vin));
                 // Update all channels from the updated data
                 getThing().getChannels().stream().map(Channel::getUID)
                         .filter(channelUID -> isLinked(channelUID) && !LAST_TRIP_GROUP.equals(channelUID.getGroupId()))
                         .forEach(channelUID -> {
-                            State state = getValue(channelUID.getIdWithoutGroup(), vehicleStatus, vehiclePosition);
+                            State state = getValue(channelUID.getGroupId(), channelUID.getIdWithoutGroup(),
+                                    vehicleStatus, vehiclePosition);
 
                             updateState(channelUID, state);
                         });
                 updateTrips(bridgeHandler);
             } catch (JsonSyntaxException | IOException e) {
                 logger.warn("Exception occurred during execution: {}", e.getMessage(), e);
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-                refreshJob = null;
+                ScheduledFuture<?> refreshJob = this.refreshJob;
+                if (refreshJob != null) {
+                    refreshJob.cancel(true);
+                    refreshJob = null;
+                }
```

Comment:
This should be put into its own method so it can be called during dispose as well.
```suggestion
                ScheduledFuture<?> refreshJob = this.refreshJob;
                if (refreshJob != null) {
                    refreshJob.cancel(true);
                    this.refreshJob = null;
                }
```